### ISSUE TITLE
It looks like with Zed Preview the panel state is toggled every time the editor is opened.

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/community#2406]
It looks like with Zed Preview the panel state is toggled every time the editor is opened.

## Environment

Zed: v0.119.8 (Zed Preview)
OS: macOS 14.2.1
Memory: 16 GiB
Architecture: aarch64

https://github.com/zed-industries/community/assets/16290753/72b29b5a-32f8-4da4-bc96-4795571df848

_Originally posted by @Moshyfawn in https://github.com/zed-industries/community/issues/1168#issuecomment-1890847018_

**Comment by JosephTLyons** [comment]
This is something I've been dealing with too, but others on the team have said they haven't ran into it. Looks like we got the first user outside of the team experiencing it. This bug is driving me crazy!

**Comment by Moshyfawn** [comment]
Oh no, one of those floating bugs 🤦

It's important to note that in my case it's only reproducible when no files are open.

I've also found that it affects the right dock too! Note that when the editor is reopened, instead of the right dock being open, the Assistant panel is open instead.


https://github.com/zed-industries/community/assets/16290753/aa16e32f-dafb-4878-af5b-eadab2ea79a5

**Comment by Moshyfawn** [comment]
It doesn't seem to affect the bottom terminal panel or the Assistant panel specifically, even though it appears to be the same dock element.

**Comment by JosephTLyons** [comment]
For me, its the left dock every time and I have that dock set to the assistant - it always opens the assistant history.

**Comment by Moshyfawn** [comment]
> For me, its the left dock every time and I have that dock set to the assistant - it always opens the assistant history.

Fair enough. I decided to test with the default layout, as I thought it would be a bit easier to debug/explain the use case.

But yes, opening the dock on the side where the Assistant panel is currently configured will cause the Assistant history to open over the dock panel.

**Comment by JosephTLyons** [comment]
This should be fixed in [v0.119.9-pre](https://zed.dev/releases/preview/0.119.9) - please reopen if the issue persists.

**Comment by Moshyfawn** [comment]
@JosephTLyons the Chat panel is still closed for the Assistant panel to reopen.

**Comment by JosephTLyons** [comment]
@Moshyfawn - if you toggle your docks a few times and then close them all, then close and reopen the app, do any docks reappear?

**Comment by Moshyfawn** [comment]
> @Moshyfawn - if you toggle your docks a few times and then close them all, then close and reopen the app, do any docks reappear?

The docks don't reappear after opening and closing them a few times. 

Leaving the Chat panel open causes the Assistant panel to open when the editor is reopened.


https://github.com/zed-industries/community/assets/16290753/53cbfecc-c61e-4a95-b48d-2ea00504b75e

**Comment by JosephTLyons** [comment]
Let's reopen this then - sorry for the confusion here. Thanks @Moshyfawn.

**Comment by osiewicz** [comment]
@Moshyfawn where is your Terminal docked? My theory is that we're persisting the dock state (whether it's closed or not), but we always set them with following priority:
1. project panel
2. terminal panel
3. assistant panel
4. channels panel
6. chat panel
7. notification panel
If that's right, I suppose neither project panel nor terminal panel are docked to the right in your setup.

**Comment by Moshyfawn** [comment]
The terminal is docked at the bottom, as per the default configuration. The Project Panel is docked on the left for testing purposes 😋

![zed-layout](https://github.com/zed-industries/community/assets/16290753/93c9f95f-ebec-43ab-af5c-a82953e2ae28)

**Comment by ConradIrwin** [comment]
Thanks for reporting! This will be fixed in the preview release later today

**Comment by JosephTLyons** [comment]
This should be shipped in [v0.119.16-pre](https://zed.dev/releases/preview/0.119.16).

**Comment by Moshyfawn** [comment]
Can confirm 👍
